examples/question-answering/run_qa.py

```diff
@@ -360,7 +374,12 @@ def prepare_train_features(examples):
         return tokenized_examples
 
     if training_args.do_train:
-        train_dataset = datasets["train"].map(
+        if "train" not in datasets:
+            raise ValueError("--do_train requires a train dataset")
+        train_dataset = datasets["train"]
+        if data_args.max_train_samples is not None:
+            train_dataset = train_dataset.select(range(data_args.max_train_samples))
+        train_dataset = train_dataset.map(
```

Comment:
The `prepare_train_features` function will create multiple entries for each example. So we should do a second select after the preprocessing.